src/core/MOM_isopycnal_slopes.F90

```diff
@@ -202,6 +215,19 @@ subroutine calc_isoneutral_slopes(G, GV, US, h, e, tv, dt_kappa_smooth, &
                                     tv%eqn_of_state, EOSdom_u)
     endif
 
+    if (use_stanley) then
+      do i=is-1,ie+1
+        pres_h(i) = pres(i,j,K)
+        T_h(i) = 0.5*(T(i,j,k) + T(i,j,k-1))
+        S_h(i) = 0.5*(S(i,j,k) + S(i,j,k-1))
+      enddo
+      ! The second line below would correspond to arguments
+      !            drho_dS_dS, drho_dS_dT, drho_dT_dT, drho_dS_dP, drho_dT_dP, &
+      call calculate_density_second_derivs(T_h, S_h, pres_h, &
+                   scrap, scrap, drho_dT_dT_h, scrap, scrap, &
+                   is-1, ie-is+3, tv%eqn_of_state)
```

Comment:
Why is it `ie-is+3` here when further down it's `ie-is+1`?

Reply:
The loops that calculate drdiA, drdjA, etc. operate over different arrays of indices. This calculation should work for indices is-1 to ie+1, i.e., over ie-is+3 points, while the one below is for to ie, over ie-is+1 points.

Reply:
OK, makes sense.